### PR TITLE
[RSDK-9520] Refactor NextPointCloud -> PointCloud and add `extra` param

### PR DESCRIPTION
**Pull request by hexbabe**
Refactor camera interface: rename NextPointCloud to PointCloud and update related tests

This PR updates the camera interface by renaming the method `NextPointCloud` to `PointCloud`, reflecting a more consistent naming convention. All instances of the method across various files, including tests and implementations, have been modified accordingly. The new method signature includes the `extra` parameter to match the proto API, enhancing flexibility and aligning the interfaces.

Changes include:
- Updated method definitions and calls in `camera.go`, `client.go`, `videosourcewrappers.go`, and other related files.
- Adjusted test cases to use the new method name and signature.

This refactor improves code clarity and prepares the codebase for further updates to point cloud processing.

### CODE REVIEW

File: components/camera/camera.go [hunk 1]
```diff
@@ -140,9 +140,9 @@ type VideoSource interface {
 	// that may have a MIME type hint dictated in the context via gostream.WithMIMETypeHint.
 	Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error)
 
-	// NextPointCloud returns the next immediately available point cloud, not necessarily one
+	// PointCloud returns the next immediately available point cloud, not necessarily one
 	// a part of a sequence. In the future, there could be streaming of point clouds.
-	NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error)
+	PointCloud(ctx context.Context, extra map[string]interface{}) (pointcloud.PointCloud, error)
```
Comment (randhid) [inline components/camera/camera.go:145]: Bro... I just realized this thing has a mimetype and returns bytes too....
https://github.com/viamrobotics/api/blob/a97363bad4477f2a2320c28be0d12447d9a81f89/proto/viam/component/camera/v1/camera.proto#L107

Comment (hexbabe) [reply components/camera/camera.go:145]: no.

Comment (hexbabe) [reply components/camera/camera.go:145]: noooooooooo

Comment (seanavery) [reply components/camera/camera.go:145]: We are ok to handle `mimetype` inclusion in another PR? There are no consequences wrt data capture by adding in `extra` to the interface method here?

Comment (randhid) [reply components/camera/camera.go:145]: > We are ok to handle mimetype inclusion in another PR?

If we are going to break the same interface I'd rather not do it twice.

**Pull request closed without merging**